src/xml_parser_main_metadata_together.c

```diff
@@ -370,30 +371,41 @@ cr_PkgIterator_new(const char *primary_path,
     cbdata->newpkgcb_data = newpkgcb_data;
 
     new_iter->tmp_err = NULL;
+    int do_files = 0;
 
     GError* tmp_err = new_iter->tmp_err;
     new_iter->primary_f = cr_open(primary_path, CR_CW_MODE_READ, CR_CW_AUTO_DETECT_COMPRESSION, &tmp_err);
+    cbdata->loadingflags |= CR_PACKAGE_LOADED_PRI;
     if (tmp_err) {
         g_propagate_prefixed_error(err, tmp_err, "Cannot open %s: ", primary_path);
         cr_PkgIterator_free(new_iter, err);
        return NULL;
     }
-    new_iter->filelists_f = cr_open(filelists_path, CR_CW_MODE_READ, CR_CW_AUTO_DETECT_COMPRESSION, &tmp_err);
-    if (tmp_err) {
-        g_propagate_prefixed_error(err, tmp_err, "Cannot open %s: ", filelists_path);
-        cr_PkgIterator_free(new_iter, err);
-        return NULL;
+    if (new_iter->filelists_path) {
+        new_iter->filelists_f = cr_open(filelists_path, CR_CW_MODE_READ, CR_CW_AUTO_DETECT_COMPRESSION, &tmp_err);
+        cbdata->loadingflags |= CR_PACKAGE_LOADED_FIL;
```

Comment:
If it's too confusing to reuse the same name for the value and the bitfield variants for a slightly different purpose, I can change that.

cbdata->loadingflags == which flags need to be present for the package to be considered "completed"
pkg->loadingflags == which files HAVE been loaded

Reply:
I think that it is fine.